### PR TITLE
Handle scroll position

### PR DESCRIPTION
**Pull request by Robinnnnn**
- [x] Maintain scroll position when user clicks "back" in browser
- [x] Maintain scroll position when user clicks "back to gallery" in NFT detail page
- [x] Forget scroll position when moving forward (e.g. edit gallery => profile)

### CODE REVIEW

File: src/components/FadeTransitioner/FadeTransitioner.tsx [hunk 1]
```diff
@@ -30,10 +31,16 @@ const transitionGroupStyles = { minHeight: fullPageHeightWithoutFooter };
...
+  const { handleNavigationScrollPosition } = useNavigationContext();
+
   return (
     <TransitionGroup style={transitionGroupStyles}>
-      <CSSTransition key={nodeKey} timeout={timeoutConfig} classNames="fade">
+      <CSSTransition
+        key={locationKey}
+        timeout={timeoutConfig}
+        classNames="fade"
+        onExit={handleNavigationScrollPosition}>
```
Comment (Robinnnnn) [inline src/components/FadeTransitioner/FadeTransitioner.tsx:43]: main change is passing this handler into `onExit`

File: src/contexts/navigation/NavigationContext.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    return removeListener;
+  }, []);
+
+  // keeps track of history length since visiting the app
+  const visitedPagesLength = useRef(0);
+  useEffect(() => {
+    visitedPagesLength.current++;
+  }, [locationKey]);
+
+  const getVisitedPagesLength = useCallback(() => visitedPagesLength.current, []);
```
Comment (Robinnnnn) [inline src/contexts/navigation/NavigationContext.tsx:46]: we can theoretically do this by simply calling `window.history.length`, but this number includes pages the user has browsed **before** arriving on the page

File: src/contexts/navigation/NavigationContext.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import { globalHistory } from '@reach/router';
```
Comment (kaitoo1) [inline src/contexts/navigation/NavigationContext.tsx:1]: love contexts

File: src/scenes/NftDetailPage/NftDetailPage.tsx [hunk 4]
```diff
@@ -8,22 +8,37 @@ import ActionText from 'components/core/ActionText/ActionText';
...
   const handleBackClick = useCallback(() => {
-    // TODO this works but leaves trailing slash
-    // navigate('..');
-    void navigate(`/${window.location.pathname.split('/')[1]}`);
-  }, []);
+    const visitedPagesLength = getVisitedPagesLength();
+
+    // if the user arrived on the page via direct link, send them to the
+    // owner's profile page (since there is no "previous page")
+    if (visitedPagesLength === 1) {
```
Comment (kaitoo1) [inline src/scenes/NftDetailPage/NftDetailPage.tsx:30]: love this

**Review by kaitoo1 — APPROVED** [review]

**Pull request merged**